lib/keymap-helpers.nix

```diff
@@ -85,9 +81,10 @@ rec {
   mkMapOptionSubmodule =
     {
       defaults ? { },
-      hasKey ? false,
-      hasAction ? false,
-      rawAction ? true,
+      # key and action can be true/false to enable/disable adding the option,
+      # or an attrset to enable the action and add/override mkOption args.
+      key ? true,
+      action ? true,
```

Comment:
Still not a perfect function signature, but at least we have unobstructed access to some of the underlying `mkOption` calls now :thinking: 

A `mode` argument makes sense too, though `mkModeOption` is currently sitting in the way of `mkOption`. Same for `options.*`.